src/python/pants/option/config.py

```diff
@@ -456,6 +463,16 @@ def options(self, section: str) -> List[str]:
         )
         return result
 
+    def _maybe_deprecated_default(self, option: str) -> None:
+        matched = option == "pants_supportdir"
+        value = self.defaults[option] if matched else None
+        deprecated_conditional(
+            lambda: matched,
+            "2.8.0.dev0",
+            "The `pants_supportdir` default value in `pants.toml`",
+            hint=f"Replace use of the variable with the literal: {repr(value)}.",
+        )
```

Comment:
This isn't causing a bunch of deprecation messages? If not, brilliant! I tried several times over the past year to do this change and couldn't figure it out.

Reply:
Yea, took some figuring out. Essentially: we can't remove putting the entry in the `DEFAULT` section (or reading it from there) because the value is actually consumed by options parsing (which is very circular... not entirely sure how that is working).

But we _can_ deprecate interpolating it... and that's the only reason it's present in `pants.toml` anyway. When we remove that deprecation, the other deprecation will begin.